Lombiq.Hosting.Tenants.Maintenance/Maintenance/ElasticsearchIndices/Startup.cs

```diff
@@ -0,0 +1,162 @@
+using Elasticsearch.Net;
+using Lombiq.HelpfulLibraries.OrchardCore.Mvc;
+using Lombiq.Hosting.Tenants.Maintenance.Constants;
+using Lombiq.Hosting.Tenants.Maintenance.Services;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Nest;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Configuration;
+using OrchardCore.Modules;
+using OrchardCore.Search.Elasticsearch.Core.Models;
+using OrchardCore.Search.Elasticsearch.Core.Services;
+using System;
+using System.Linq;
+
+namespace Lombiq.Hosting.Tenants.Maintenance.Maintenance.ElasticsearchIndices;
+
+[Feature(FeatureNames.DeleteOrRebuildElasticsearchIndices)]
+public class DeleteElasticsearchIndicesStartup : StartupBase
+{
+    private readonly IShellConfiguration _shellConfiguration;
+
+    public DeleteElasticsearchIndicesStartup(IShellConfiguration shellConfiguration) => _shellConfiguration = shellConfiguration;
+
+    public override void ConfigureServices(IServiceCollection services)
+    {
+        services.BindAndConfigureSection<ElasticsearchIndicesMaintenanceOptions>(
+            _shellConfiguration,
+            "Lombiq_Hosting_Tenants_Maintenance:ElasticsearchIndicesOptions");
+
+        services.AddScoped<IMaintenanceProvider, DeleteElasticsearchIndicesMaintenanceProvider>();
+        services.AddScoped<IMaintenanceProvider, RebuildElasticsearchIndicesMaintenanceProvider>();
+    }
+}
+
+[Feature(FeatureNames.DeleteElasticsearchIndicesBeforeSetup)]
+public class DeleteElasticsearchIndicesBeforeSetupStartup : StartupBase
+{
+    private readonly IShellConfiguration _shellConfiguration;
+    private readonly ShellSettings _shellSettings;
+
+    public DeleteElasticsearchIndicesBeforeSetupStartup(IShellConfiguration shellConfiguration, ShellSettings shellSettings)
+    {
+        _shellConfiguration = shellConfiguration;
+        _shellSettings = shellSettings;
+    }
+
+    public override void ConfigureServices(IServiceCollection services)
+    {
+        services.BindAndConfigureSection<ElasticsearchIndicesMaintenanceOptions>(
+            _shellConfiguration,
+            "Lombiq_Hosting_Tenants_Maintenance:ElasticsearchIndicesOptions");
+
+        // After the setup, the Elasticsearch module can be loaded the regular way.
+        if (!_shellSettings.IsUninitialized()) return;
+
+        // This is necessary because the Elasticsearch module can't be enabled the regular way if this module is added
+        // as a setup feature, otherwise you get a ContentsAdminList shape missing exception on the admin dashboard. For
+        // more info see:
```

Comment:
Did you miss a link here?

Reply:
Kind of no, I rewrote the documentation here.